timezone_converter/main.py

```diff
@@ -30,8 +44,11 @@ def build_parser() -> argparse.ArgumentParser:
     parser.add_argument(
         '-l',
         '--list',
-        action='store_true',
-        help='show all available timezones',
+        nargs='?',
+        type=_single_letter,
+        const='all',
+        metavar='LETTER',
+        help='show all available timezones or timezones beginning with specified letter groups',
```

Comment:
Here we should use an easier to distinguish value for `const`, like `None`, but to be able to get the value by circonventing argparser's checks, we need to use `[None]`.

We also should use a shorter string to make `flake8` happy and I think that this one can be a good fit, as it is descriptive and concise:

> show all timezones or only those that start with specific letters